internal/controller/pkg/revision/reconciler.go

```diff
@@ -388,6 +391,15 @@ func (r *Reconciler) Reconcile(ctx context.Context, req reconcile.Request) (reco
 	}
 
 	pkgMeta, _ := xpkg.TryConvert(pkg.GetMeta()[0], &pkgmetav1.Provider{}, &pkgmetav1.Configuration{})
+
+	pr.SetAnnotations(pkgMeta.(metav1.ObjectMetaAccessor).GetObjectMeta().GetAnnotations())
+	if err := r.client.Update(ctx, pr); err != nil {
+		r.record.Event(pr, event.Warning(reasonSync, err))
```

Comment:
Might be nice to provide some additional context as to why we are attempting an update :+1: 
```suggestion
		r.record.Event(pr, event.Warning(reasonSync, errors.Wrap(err, errUpdateAnnotations)))
```